src/extension.js

```diff
@@ -58,6 +58,10 @@ let OSKIndicator = GObject.registerClass(
       this.connect("touch-event", function () {
         toggleOSK();
       });
+      
+      settings.connect("changed::toggle", function () {
+        toggleOSK();
+      });
```

Comment:
Please move this into `enable()` method, so it would be next to other `settings.connect(...)` calls.